### PR TITLE
tests: fix for failover test on how logs are checked

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Otherwise logs used are since last reboot.

error:
https://api.travis-ci.org/v3/job/460209244/log.txt

### CODE REVIEW

File: tests/main/failover/task.yaml [hunk 1]
```diff
@@ -87,10 +90,17 @@ execute: |
...
+        if check_journalctl_log "Waiting for system reboot"; then
+            echo "Already waiting for system reboot, exiting..."
+            exit 1
+        fi
+
         # install new target snap
         snap install --dangerous failing.snap
 
-        while ! journalctl -b -u snapd|grep -q "Waiting for system reboot" ; do sleep 1 ; done
+        while ! check_journalctl_log "Waiting for system reboot"; do
```
Comment (bboozzoo) [inline tests/main/failover/task.yaml]: Would be a bit easier to understand with the comment:
```
# use journalctl wrapper to grep only the logs collected while the test is running
```

File: tests/main/failover/task.yaml [hunk 2]
```diff
@@ -87,10 +90,17 @@ execute: |
...
 
+        if check_journalctl_log "Waiting for system reboot"; then
+            echo "Already waiting for system reboot, exiting..."
+            exit 1
+        fi
+
         # install new target snap
         snap install --dangerous failing.snap
 
-        while ! journalctl -b -u snapd|grep -q "Waiting for system reboot" ; do sleep 1 ; done
```
Comment (bboozzoo) [inline tests/main/failover/task.yaml]: Wonder why `journalctl -b` didn't work as expected. In theory those are the logs from current boot only.

**Comment by mvo5** [comment]
Nice, this eliminates a class of spurious errors we saw, right?

**Comment by sergiocazzolato** [comment]
@bboozzoo, thanks for the review!! could you please take a look again?

**Pull request merged**